app/components/log-drain-actions/component.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+
+  logDrain: null,
+
+  sendError(e, message) {
+    message = Ember.get(e, 'responseJSON.message') || message;
+    this.sendAction('failedAction', message);
+  },
+
+  actions: {
+    restart(){
+      let message = `Successfully restarted ${this.logDrain.get('handle')}`;
+      let errorMessage = `There was an error restarting ${this.logDrain.get('handle')}.`;
+      let logDrain = this.logDrain;
+      let component = this;
+      var op = this.get('store').createRecord('operation', {
+        type: 'configure',
+        logDrain: this.logDrain
+      });
+      op.save().then(function(){
+        logDrain.set('status', 'provisioning');
+        logDrain.save().then(function() {
+          component.sendAction('completedAction', message);
+        }).catch( (e) => { component.sendError(e, errorMessage); });
+      }).catch( (e) => { component.sendError(e, errorMessage); });
+    },
+
+    deprovision(){
+      let message = `Successfully deprovisioned ${this.logDrain.get('handle')}`;
+      let errorMessage = `There was an error restarting ${this.logDrain.get('handle')}.`;
+      let logDrain = this.logDrain;
```

Comment:
I created local references for the closures around the `.then`s. Since I have `component`, I can use `component.logDrain`. Maybe I need to try again... `this` had the wrong context.